orchagent/portsorch.cpp

```diff
@@ -2900,6 +2989,15 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         SWSS_LOG_NOTICE("Set port %s AutoNeg from %d to %d", alias.c_str(), p.m_autoneg, an);
                         p.m_autoneg = an;
                         m_portList[alias] = p;
+                        m_portStateTable.hdel(p.m_alias, "rmt_adv_speeds");
+                        if (an > 0)
+                        {
+                            m_port_state_poll[p.m_alias] |= PORT_STATE_POLL_AN;
+                        }
+                        else
+                        {
+                            m_port_state_poll[p.m_alias] &= ~PORT_STATE_POLL_AN;
```

Comment:
How about remove the port from m_port_state_poll if it no longer requires polling? The benefit is that we can stop the timer if  m_port_state_poll  is empty.

Reply:
Sure, will do